### PR TITLE
util module: extension and span utils

### PR DESCRIPTION
**Pull request by RingerJK**
## Description

Please include a brief summary of the change and which issue is fixed (e.g., fixes [#issue](link))

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Please describe the PR goals. Just the stuff needed to implement the fix / feature and a simple rationale. It could contain many check points if needed

### Implementation

Please include all the relevant things implemented and also rationale, clarifications / disclaimers etc. related to the approach used. It could be as self code companion comments

## Screenshots or Gifs

Please include all the media files to give some context about what's being implemented or fixed. It's not mandatory to upload screenshots or gifs, but for most of the cases it becomes really handy to get into the scope of the feature / bug being fixed and also it's REALLY useful for UI related PRs screenshot gif

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

**Comment by Lebedinsky** [comment]
@RingerJK CircleCI failed, please take a 👀 what is happened

**Comment by RingerJK** [comment]
@mapbox/navigation-android PTAL

File: libdirections-hybrid/build.gradle [hunk 1]
```diff
@@ -19,6 +19,7 @@ dependencies {
     api(project(':libnavigation-base'))
     implementation(project(":libdirections-onboard"))
     implementation(project(":libdirections-offboard"))
+    implementation project(':libnavigation-util')
```
Comment (LukasPaczos) [inline libdirections-hybrid/build.gradle]: Let's add the dependency only when the module will actually need any of the tools from `libnavigation-utils`. Same for all other modules.

Comment (RingerJK) [reply libdirections-hybrid/build.gradle]: good catch. done 👍

File: libdirections-offboard/build.gradle [hunk 2]
```diff
@@ -15,6 +15,7 @@ android {
 }
 
 dependencies {
+    implementation project(':libnavigation-util')
```
Comment (LukasPaczos) [inline libdirections-offboard/build.gradle]: Also unused at this point, right?

Comment (RingerJK) [reply libdirections-offboard/build.gradle]: no, here it'll be used. My next PR will change this module

Comment (LukasPaczos) [reply libdirections-offboard/build.gradle]: At this point unused though :upside_down_face:

**Pull request merged**